testsuite/performance/tests/src/test/scala/keycloak/DefaultSimulation.scala

```diff
@@ -54,100 +49,94 @@ class DefaultSimulation extends Simulation {
-    }
-
-    // Successful login
-    .exec(http("Browser posts correct credentials")
-      .post("${login-form-uri}")
-      .headers(UI_HEADERS)
-      .formParam("username", "${username}")
-      .formParam("password", "${password}")
-      .formParam("login", "Log in")
-      .check(status.is(302), header("Location").saveAs("login-redirect")))
-    .exitHereIfFailed
-
-
-    // Now act as client adapter - exchange code for keys
-    .exec(oauth("Adapter exchanges code for tokens")
-      .authorize("${login-redirect}",
-      session => List(new Cookie("OAuth_Token_Request_State", session("state").as[String], 0, null, null)))
-      .authServerUrl("${keycloakServer}")
-      .resource("${clientId}")
-      .clientCredentials("${secret}")
-      .realm("${realm}")
-    //.realmKey(Loader.realmRepresentation.getPublicKey)
-    )
-
-    // Refresh token several times
-    .asLongAs(s => downCounterAboveZero(s, "refreshTokenCount")) {
-      pause(TestConfig.refreshTokenPeriod, Normal(TestConfig.refreshTokenPeriod * 0.2))
-      .exec(oauth("Adapter refreshes token").refresh())
-    }
+        .check(status.is(302), header("Location").saveAs("login-redirect")))
+  .exitHereIfFailed
+
+
+  // Now act as client adapter - exchange code for keys
+  .exec(oauth("Adapter exchanges code for tokens")
+        .authorize("${login-redirect}",
+                   session => List(new Cookie("OAuth_Token_Request_State", session("state").as[String], 0, null, null)))
+        .authServerUrl("${keycloakServer}")
+        .resource("${clientId}")
+        .clientCredentials("${secret}")
+        .realm("${realm}")
+        //.realmKey(Loader.realmRepresentation.getPublicKey)
+  )
+
+  // Refresh token several times
+  .asLongAs(s => downCounterAboveZero(s, "refreshTokenCount")) {
+    pause(TestConfig.refreshTokenPeriod, Normal(TestConfig.refreshTokenPeriod * 0.2))
+    .exec(oauth("Adapter refreshes token").refresh())
+  }
 
-    // Logout
-    .pause(TestConfig.userThinkTime, Normal(TestConfig.userThinkTime * 0.2))
-    .exec(http("Browser logout")
-      .get(LOGOUT_ENDPOINT)
-      .headers(UI_HEADERS)
-      .queryParam("redirect_uri", "${appUrl}")
-      .check(status.is(302), header("Location").is("${appUrl}")))
+  // Logout
+  .pause(TestConfig.userThinkTime, Normal(TestConfig.userThinkTime * 0.2))
+  .exec(http("Browser logout")
+        .get(LOGOUT_ENDPOINT)
+        .headers(UI_HEADERS)
+        .queryParam("redirect_uri", "${appUrl}")
+        .check(status.is(302), header("Location").is("${appUrl}")))
 
   val usersScenario = scenario("users")
-    .repeat(TestConfig.numOfIterations) {
-      userSession
-    }
-
-  setUp(usersScenario.inject( {
-    if (TestConfig.rampUpPeriod > 0) {
-      rampUsers(TestConfig.runUsers) over TestConfig.rampUpPeriod
-    } else {
-      atOnceUsers(TestConfig.runUsers)
-    }
-  }).protocols(httpDefault))
-
-
+  .asLongAs(s => rampDownPeriodNotReached(), null, TestConfig.rampDownASAP) {
```

Comment:
I replaced the `.repeat(TestConfig.numOfIterations)` loop (which has a fixed number of iterations) with `.asLongAs(s => rampDownPeriodNotReached() ...)` loop which checks if the `rampUpPeriod + steadyLoadPeriod` has expired, and keeps repeating the scenario otherwise.

When the `rampDownASAP` parameter is true the check will be done after each scenario step, otherwise it will be done at the end of each scenario iteration.